### DISCUSSION TITLE
Principal Curvatures and Directions

### DISCUSSION THREAD

**Discussion by royvelich** [cnr-isti-vclab/PyMeshLab#352] [Q&A]
Hi,
I see that there are two filters that can calculate the principal curvatures at each vertex - `compute_curvature_and_color_apss_per_vertex` and `compute_curvature_and_color_rimls_per_vertex`, but how can I get back the curvature values form the `Mesh` object? I cannot find any relevant method that returns back those values.

Also, on the other hand, I see that there are two methods in the `Mesh` object that return the principal directions at each face -  `face_curvature_principal_dir1_matrix` and `face_curvature_principal_dir2_matrix`. However, there is no filter that calculates them. Is it possible to calculate them?

Thanks!

**Comment by keblicuey** [comment]
Hey, I have the same question with you. Did you figure it out?
